### PR TITLE
Added input sanitation to policy

### PR DESCRIPTION
**Pull request by sanket1729**
Addresses #48 , #47 , #46 , #45 , #44 

Thanks @practicalswift for pointing these out

### CODE REVIEW

**Comment by practicalswift** [comment]
Thanks for fixing and crediting!

If you `s/Addresses/Fixes/` then GitHub will automatically close the referenced issues when this PR is merged :)

File: src/policy/concrete.rs [hunk 1]
```diff
@@ -57,10 +57,55 @@ pub enum Policy<Pk: MiniscriptKey> {
...
+    /// `And` fragments only support two args
+    NonBinaryArgAnd,
+    /// `Or` fragments only support two args
+    NonBinaryArgOr,
+    /// `Thresh` fragment can only have `1<=k<=n`
+    IncorrectThresh,
+    /// `older` or `after` fragment can only have `n >= 0`
+    ZeroTime,
+    /// `after` fragment can only have ` n < 2^31`
+    TimeToofar,
```
Comment (practicalswift) [inline src/policy/concrete.rs]: `TimeTooFar` to be consistent with the capitalisation of the other errors?

File: src/policy/concrete.rs [hunk 2]
```diff
@@ -57,10 +57,55 @@ pub enum Policy<Pk: MiniscriptKey> {
...
+            }
+            PolicyError::NonBinaryArgOr => f.write_str("Or policy fragment must take 2 arguments"),
+            PolicyError::IncorrectThresh => {
+                f.write_str("Threshold k must be greater than 0 and less than n")
+            }
+            PolicyError::TimeToofar => {
+                f.write_str("Relative/Absolute time must be less than 2^31; n < 2^31")
+            }
+            PolicyError::ZeroTime => {
+                f.write_str("Relative/Absolute time must be greater than 0; n>=0")
```
Comment (practicalswift) [inline src/policy/concrete.rs]: Should be `n  > 0`?

Nit: Could replace "Relative/Absolute time must be …" with "Time must be …". Applies above too.

**Comment by apoelstra** [comment]
ACK except for the listed nits

**Comment by sanket1729** [comment]
Fixes #48 , #47 , #46 , #45 , #44

**Pull request merged**